internal/pkg/cli/task_run.go

```diff
@@ -259,16 +261,54 @@ func (o *runTaskOpts) Validate() error {
 	return nil
 }
 
+func (o *runTaskOpts) validateRunnerTypeOptions() error {
+	// Subnets cannot be specified with default.
+	if o.defaultEnv && o.subnets != nil {
+		return fmt.Errorf("cannot specify both subnets and default")
+	}
+
+	// Application and environment cannot be specified with default.
+	if o.appName != "" && o.defaultEnv {
+		return fmt.Errorf("cannot specify both application and default")
+	}
+
+	if o.env != "" && o.defaultEnv {
+		return fmt.Errorf("cannot specify both environment and default")
+	}
+
+	// Application and environment cannot be specified with subnets or security groups.
+	if o.appName != "" && o.subnets != nil {
+		return fmt.Errorf("cannot specify both subnets and application")
+	}
+
+	if o.appName != "" && o.securityGroups != nil {
+		return fmt.Errorf("cannot specify both security groups and application")
+	}
+
+	if o.env != "" && o.subnets != nil {
+		return fmt.Errorf("cannot specify both subnets and environment")
+	}
+
+	if o.env != "" && o.securityGroups != nil {
+		return fmt.Errorf("cannot specify both security groups and environment")
+	}
+
+	return nil
+}
+
 // Ask prompts the user for any required or important fields that are not provided.
 func (o *runTaskOpts) Ask() error {
 	if err := o.askTaskGroupName(); err != nil {
 		return err
 	}
-	if err := o.askAppName(); err != nil {
-		return err
-	}
-	if err := o.askEnvName(); err != nil {
-		return err
+
+	if !o.defaultEnv && o.subnets == nil {
```

Comment:
maybe add a comment here saying "if using copilot environment". However, should we make "using copilot env" as default or we should make "using default setting" as default? I'm more inclined to the latter since that's what "default" should be...

Reply:
jes, or creating a separate method that contains this boolean logic:
```go
func (o *runTaskOpts) useCopilotEnv() bool {
    return !o.defaultEnv && o.subnets == nil
}
```